PR to add  and correct doxygen docblocks to emcsfc_snow2mdl.fd/*.f

GeorgeVandenberghe-NOAA opened the pull request:
Add and correct doxygen docblocks to emcsfc_snow2mdl.fd/*.f.

Part of #291

GeorgeVandenberghe-NOAA commented:
Something happened to this PR description and it got lost so I'm adding it here.   It is adding some changes to sorc/emcsfc_snow2mdl.fd/model_grid.f

GeorgeVandenberghe-NOAA commented:
There is a grib_utils.f conflict.   Don't add that in.  The one in develop is already good.  Mine is to be discarded.

GeorgeGayno-NOAA commented:
> There is a grib_utils.f conflict. Don't add that in. The one in develop is already good. Mine is to be discarded.

Your branch is likely out of date with 'develop'.

GeorgeVandenberghe-NOAA commented:
ANd how do I sync back to develop.

Ed gave me this. It fails
git clone git@github.com:GeorgeVandenberghe-NOAA/UFS_UTILS.git
cd UFS_UTILS
git add remote upstream git@github.com:NOAA-EMC/UFS_UTILS.git
git fetch upstream


+ git clone git@github.com:GeorgeVandenberghe-NOAA/UFS_UTILS.git
Cloning into 'UFS_UTILS'...
remote: Enumerating objects: 533, done.
remote: Counting objects: 100% (533/533), done.
remote: Compressing objects: 100% (282/282), done.
remote: Total 19287 (delta 341), reused 409 (delta 251), pack-reused 18754
Receiving objects: 100% (19287/19287), 43.86 MiB | 29.09 MiB/s, done.
Resolving deltas: 100% (12951/12951), done.
+ cd UFS_UTILS
+ git add remote upstream git@github.com:NOAA-EMC/UFS_UTILS.git
fatal: pathspec 'remote' did not match any files
+ git fetch upstream
fatal: 'upstream' does not appear to be a git repository
fatal: Could not read from remote repository.






On Tue, Feb 23, 2021 at 2:19 PM GeorgeGayno-NOAA <notifications@github.com>
wrote:

> There is a grib_utils.f conflict. Don't add that in. The one in develop is
> already good. Mine is to be discarded.
>
> Your branch is likely out of date with 'develop'.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/UFS_UTILS/pull/348#issuecomment-784447902>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ANDS4FRTF4O63FVLFQNJ2LTTAP5TBANCNFSM4YC6U2SA>
> .
>


-- 

George W Vandenberghe

*IMSG* at NOAA/NWS/NCEP/EMC

5830 University Research Ct., Rm. 2141

College Park, MD 20740

George.Vandenberghe@noaa.gov

301-683-3769(work)  3017751547(cell)

GeorgeGayno-NOAA commented:
Try this: 3. Checking out code and making changes · NOAA-EMC/UFS_UTILS Wiki
(github.com)
<https://github.com/NOAA-EMC/UFS_UTILS/wiki/3.-Checking-out-code-and-making-changes>

On Tue, Feb 23, 2021 at 3:24 PM GeorgeVandenberghe-NOAA <
notifications@github.com> wrote:

> ANd how do I sync back to develop.
>
> Ed gave me this. It fails
> git clone git@github.com:GeorgeVandenberghe-NOAA/UFS_UTILS.git
> cd UFS_UTILS
> git add remote upstream git@github.com:NOAA-EMC/UFS_UTILS.git
> git fetch upstream
>
>
> + git clone git@github.com:GeorgeVandenberghe-NOAA/UFS_UTILS.git
> Cloning into 'UFS_UTILS'...
> remote: Enumerating objects: 533, done.
> remote: Counting objects: 100% (533/533), done.
> remote: Compressing objects: 100% (282/282), done.
> remote: Total 19287 (delta 341), reused 409 (delta 251), pack-reused 18754
> Receiving objects: 100% (19287/19287), 43.86 MiB | 29.09 MiB/s, done.
> Resolving deltas: 100% (12951/12951), done.
> + cd UFS_UTILS
> + git add remote upstream git@github.com:NOAA-EMC/UFS_UTILS.git
> fatal: pathspec 'remote' did not match any files
> + git fetch upstream
> fatal: 'upstream' does not appear to be a git repository
> fatal: Could not read from remote repository.
>
>
>
>
>
>
> On Tue, Feb 23, 2021 at 2:19 PM GeorgeGayno-NOAA <notifications@github.com
> >
> wrote:
>
> > There is a grib_utils.f conflict. Don't add that in. The one in develop
> is
> > already good. Mine is to be discarded.
> >
> > Your branch is likely out of date with 'develop'.
> >
> > —
> > You are receiving this because you authored the thread.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/NOAA-EMC/UFS_UTILS/pull/348#issuecomment-784447902>,
> > or unsubscribe
> > <
> https://github.com/notifications/unsubscribe-auth/ANDS4FRTF4O63FVLFQNJ2LTTAP5TBANCNFSM4YC6U2SA
> >
> > .
> >
>
>
> --
>
> George W Vandenberghe
>
> *IMSG* at NOAA/NWS/NCEP/EMC
>
> 5830 University Research Ct., Rm. 2141
>
> College Park, MD 20740
>
> George.Vandenberghe@noaa.gov
>
> 301-683-3769(work) 3017751547(cell)
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/UFS_UTILS/pull/348#issuecomment-784486537>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AMSYBTF7I6CIB6DGJTTTC73TAQFITANCNFSM4YC6U2SA>
> .
>


-- 
George Gayno
IMSG at NOAA/NWS/NCEP/EMC
5830 University Research Ct.,  Rm. 2882
College Park, MD  20740
George.Gayno@noaa.gov
301-683-3658

GeorgeVandenberghe-NOAA commented:
That worked but I ran into conflicts.  The conflicts were resolvable this
time but the general process of fixing conflicts is something else I need
more experience with.   I have a PR #348 with the conflict included.  I've
repaired it locally and pushed it back to the gwv branch of my fork from
which the PR is issued    How do I fix that conflict in the PR?



On Tue, Feb 23, 2021 at 3:28 PM GeorgeGayno-NOAA <notifications@github.com>
wrote:

> Try this: 3. Checking out code and making changes · NOAA-EMC/UFS_UTILS Wiki
> (github.com)
> <
> https://github.com/NOAA-EMC/UFS_UTILS/wiki/3.-Checking-out-code-and-making-changes
> >
>
> On Tue, Feb 23, 2021 at 3:24 PM GeorgeVandenberghe-NOAA <
> notifications@github.com> wrote:
>
> > ANd how do I sync back to develop.
> >
> > Ed gave me this. It fails
> > git clone git@github.com:GeorgeVandenberghe-NOAA/UFS_UTILS.git
> > cd UFS_UTILS
> > git add remote upstream git@github.com:NOAA-EMC/UFS_UTILS.git
> > git fetch upstream
> >
> >
> > + git clone git@github.com:GeorgeVandenberghe-NOAA/UFS_UTILS.git
> > Cloning into 'UFS_UTILS'...
> > remote: Enumerating objects: 533, done.
> > remote: Counting objects: 100% (533/533), done.
> > remote: Compressing objects: 100% (282/282), done.
> > remote: Total 19287 (delta 341), reused 409 (delta 251), pack-reused
> 18754
> > Receiving objects: 100% (19287/19287), 43.86 MiB | 29.09 MiB/s, done.
> > Resolving deltas: 100% (12951/12951), done.
> > + cd UFS_UTILS
> > + git add remote upstream git@github.com:NOAA-EMC/UFS_UTILS.git
> > fatal: pathspec 'remote' did not match any files
> > + git fetch upstream
> > fatal: 'upstream' does not appear to be a git repository
> > fatal: Could not read from remote repository.
> >
> >
> >
> >
> >
> >
> > On Tue, Feb 23, 2021 at 2:19 PM GeorgeGayno-NOAA <
> notifications@github.com
> > >
> > wrote:
> >
> > > There is a grib_utils.f conflict. Don't add that in. The one in develop
> > is
> > > already good. Mine is to be discarded.
> > >
> > > Your branch is likely out of date with 'develop'.
> > >
> > > —
> > > You are receiving this because you authored the thread.
> > > Reply to this email directly, view it on GitHub
> > > <https://github.com/NOAA-EMC/UFS_UTILS/pull/348#issuecomment-784447902
> >,
> > > or unsubscribe
> > > <
> >
> https://github.com/notifications/unsubscribe-auth/ANDS4FRTF4O63FVLFQNJ2LTTAP5TBANCNFSM4YC6U2SA
> > >
> > > .
> > >
> >
> >
> > --
> >
> > George W Vandenberghe
> >
> > *IMSG* at NOAA/NWS/NCEP/EMC
> >
> > 5830 University Research Ct., Rm. 2141
> >
> > College Park, MD 20740
> >
> > George.Vandenberghe@noaa.gov
> >
> > 301-683-3769(work) 3017751547(cell)
> >
> > —
> > You are receiving this because you commented.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/NOAA-EMC/UFS_UTILS/pull/348#issuecomment-784486537>,
> > or unsubscribe
> > <
> https://github.com/notifications/unsubscribe-auth/AMSYBTF7I6CIB6DGJTTTC73TAQFITANCNFSM4YC6U2SA
> >
> > .
> >
>
>
> --
> George Gayno
> IMSG at NOAA/NWS/NCEP/EMC
> 5830 University Research Ct., Rm. 2882
> College Park, MD 20740
> George.Gayno@noaa.gov
> 301-683-3658
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/UFS_UTILS/pull/348#issuecomment-784488852>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ANDS4FTSHKGRS7YYR72ZX4TTAQFX7ANCNFSM4YC6U2SA>
> .
>


-- 

George W Vandenberghe

*IMSG* at NOAA/NWS/NCEP/EMC

5830 University Research Ct., Rm. 2141

College Park, MD 20740

George.Vandenberghe@noaa.gov

301-683-3769(work)  3017751547(cell)

GeorgeGayno-NOAA commented:
> That worked but I ran into conflicts. The conflicts were resolvable this time but the general process of fixing conflicts is something else I need more experience with. I have a PR #348 with the conflict included. I've repaired it locally and pushed it back to the gwv branch of my fork from which the PR is issued How do I fix that conflict in the PR? On Tue, Feb 23, 2021 at 3:28 PM GeorgeGayno-NOAA <notifications@github.com> wrote:
> […](#)
> Try this: 3. Checking out code and making changes · NOAA-EMC/UFS_UTILS Wiki (github.com) < https://github.com/NOAA-EMC/UFS_UTILS/wiki/3.-Checking-out-code-and-making-changes > On Tue, Feb 23, 2021 at 3:24 PM GeorgeVandenberghe-NOAA < ***@***.***> wrote: > ANd how do I sync back to develop. > > Ed gave me this. It fails > git clone ***@***.***:GeorgeVandenberghe-NOAA/UFS_UTILS.git > cd UFS_UTILS > git add remote upstream ***@***.***:NOAA-EMC/UFS_UTILS.git > git fetch upstream > > > + git clone ***@***.***:GeorgeVandenberghe-NOAA/UFS_UTILS.git > Cloning into 'UFS_UTILS'... > remote: Enumerating objects: 533, done. > remote: Counting objects: 100% (533/533), done. > remote: Compressing objects: 100% (282/282), done. > remote: Total 19287 (delta 341), reused 409 (delta 251), pack-reused 18754 > Receiving objects: 100% (19287/19287), 43.86 MiB | 29.09 MiB/s, done. > Resolving deltas: 100% (12951/12951), done. > + cd UFS_UTILS > + git add remote upstream ***@***.***:NOAA-EMC/UFS_UTILS.git > fatal: pathspec 'remote' did not match any files > + git fetch upstream > fatal: 'upstream' does not appear to be a git repository > fatal: Could not read from remote repository. > > > > > > > On Tue, Feb 23, 2021 at 2:19 PM GeorgeGayno-NOAA < ***@***.*** > > > wrote: > > > There is a grib_utils.f conflict. Don't add that in. The one in develop > is > > already good. Mine is to be discarded. > > > > Your branch is likely out of date with 'develop'. > > > > — > > You are receiving this because you authored the thread. > > Reply to this email directly, view it on GitHub > > <[#348 (comment)](https://github.com/NOAA-EMC/UFS_UTILS/pull/348#issuecomment-784447902) >, > > or unsubscribe > > < > https://github.com/notifications/unsubscribe-auth/ANDS4FRTF4O63FVLFQNJ2LTTAP5TBANCNFSM4YC6U2SA > > > > . > > > > > -- > > George W Vandenberghe > > *IMSG* at NOAA/NWS/NCEP/EMC > > 5830 University Research Ct., Rm. 2141 > > College Park, MD 20740 > > ***@***.*** > > 301-683-3769(work) 3017751547(cell) > > — > You are receiving this because you commented. > Reply to this email directly, view it on GitHub > <[#348 (comment)](https://github.com/NOAA-EMC/UFS_UTILS/pull/348#issuecomment-784486537)>, > or unsubscribe > < https://github.com/notifications/unsubscribe-auth/AMSYBTF7I6CIB6DGJTTTC73TAQFITANCNFSM4YC6U2SA > > . > -- George Gayno IMSG at NOAA/NWS/NCEP/EMC 5830 University Research Ct., Rm. 2882 College Park, MD 20740 ***@***.*** 301-683-3658 — You are receiving this because you authored the thread. Reply to this email directly, view it on GitHub <[#348 (comment)](https://github.com/NOAA-EMC/UFS_UTILS/pull/348#issuecomment-784488852)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/ANDS4FTSHKGRS7YYR72ZX4TTAQFX7ANCNFSM4YC6U2SA> .
> -- George W Vandenberghe *IMSG* at NOAA/NWS/NCEP/EMC 5830 University Research Ct., Rm. 2141 College Park, MD 20740 George.Vandenberghe@noaa.gov 301-683-3769(work) 3017751547(cell)

If you fixed the conflict correctly, the PR should automatically update.  In this case, to fix the conflict you would probably just accept the version of grib_utils.f from 'develop'.

edwardhartnett commented:
@GeorgeVandenberghe-NOAA tell me when you are ready and I will review.

GeorgeVandenberghe-NOAA commented:
Inspect for problems and reject.  I have the same issue as grib_utils.f.
The embedded subroutine docblock is not being included in the HTML
documentation.  This is in model_grid.f

Been distracted all afternoon by a non NCEPLIBS  project meeting where I
told people I'm off the project.. deal with it, then a UFS utils meeting
concerning containers, and my son's car failed and I had to bail him out of
trouble.

On Tue, Feb 23, 2021 at 4:08 PM Edward Hartnett <notifications@github.com>
wrote:

> @GeorgeVandenberghe-NOAA <https://github.com/GeorgeVandenberghe-NOAA>
> tell me when you are ready and I will review.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/UFS_UTILS/pull/348#issuecomment-784513753>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ANDS4FW5B2DAYJ27UANLEQ3TAQKMVANCNFSM4YC6U2SA>
> .
>


-- 

George W Vandenberghe

*IMSG* at NOAA/NWS/NCEP/EMC

5830 University Research Ct., Rm. 2141

College Park, MD 20740

George.Vandenberghe@noaa.gov

301-683-3769(work)  3017751547(cell)

edwardhartnett commented on sorc/emcsfc_snow2mdl.fd/model_grid.f:
```diff
@@ -1,7 +1,7 @@
 !! @brief Read in data defining the model grid.
 !!
-!! @author gayno org: w/np2 @date 2005-dec-16
+!!@author gayno org: w/np2 @date 2005-dec-16  
```
Blank space before author tag.

edwardhartnett replied:
Blank space before author tag.

edwardhartnett commented on sorc/emcsfc_snow2mdl.fd/model_grid.f:
```diff
@@ -58,57 +58,53 @@ module model_grid
  real                           :: resol_mdl  ! in km
 
- contains
+!> 
```
No brief tag is needed for subroutines/functions. Remove it.
Text on first line, like this:

```
!> This is the brief comment.
!!
!! Now we elaborate with many sentences of additional documentation about this
!! function. We can go on and on.
!!
```

edwardhartnett replied:
Remove the brief tag (the "@brief") but leave the sentence which documents the function/subroutine.

See my example above.

edwardhartnett commented on sorc/emcsfc_snow2mdl.fd/model_grid.f:
```diff
@@ -58,57 +58,53 @@ module model_grid
+!!
+!! usage: call read_mdl_grid_info
+!!
+!!   input argument list:  n/a
```
Remove any documentation element that says n/a or none. It is not conveying information.

edwardhartnett commented on sorc/emcsfc_snow2mdl.fd/model_grid.f:
```diff
@@ -58,57 +58,53 @@ module model_grid
+!!
+!! remarks: none.
+!!
 
```
There should not be a blank like, and the documentation block should be after "contains".

GeorgeVandenberghe-NOAA replied:
Moved docblock to after the contains statement   Added doxygen comments for snowdat.f.

edwardhartnett commented:
Please also correct the name of this PR and its description. In future, do not introduce a PR with a name "Gwv" always use something descriptive to the rest of the team. Also use draft pull request for PRs you are working on, which are not ready for review.

GeorgeVandenberghe-NOAA commented:
The issues described, e.g

*!! remarks: none.
+!!

*

*There should not be a blank like, and the documentation block should be
after "contains"*.
have been corrected.


On Tue, Feb 23, 2021 at 4:27 PM Edward Hartnett <notifications@github.com>
wrote:

> *@edwardhartnett* requested changes on this pull request.
> ------------------------------
>
> In sorc/emcsfc_snow2mdl.fd/model_grid.f
> <https://github.com/NOAA-EMC/UFS_UTILS/pull/348#discussion_r581399234>:
>
> > @@ -1,7 +1,7 @@
>  !> @file
>  !! @brief Read in data defining the model grid.
>  !!
> -!! @author gayno org: w/np2 @date 2005-dec-16
> +!!@author gayno org: w/np2 @date 2005-dec-16
>
> Blank space before author tag.
> ------------------------------
>
> In sorc/emcsfc_snow2mdl.fd/model_grid.f
> <https://github.com/NOAA-EMC/UFS_UTILS/pull/348#discussion_r581400321>:
>
> > @@ -58,57 +58,53 @@ module model_grid
>   real, allocatable              :: lsmask_mdl_sav (:,:)
>   real                           :: resol_mdl  ! in km
>
> - contains
> +!>
>
> No brief tag is needed for subroutines/functions. Remove it.
> Text on first line, like this:
>
> !> This is the brief comment.
> !!
> !! Now we elaborate with many sentences of additional documentation about this
> !! function. We can go on and on.
> !!
>
> ------------------------------
>
> In sorc/emcsfc_snow2mdl.fd/model_grid.f
> <https://github.com/NOAA-EMC/UFS_UTILS/pull/348#discussion_r581400668>:
>
> > +!>
> +!! @brief read latitude, longitude, land/sea mask on the    model grid.
> +!!
> +!! @author George Gayno org: w/np2 @date 2005-dec-16
> +!!
> +!!
> +!! program history log:
> +!! 2005-dec-16  gayno    - initial version
> +!! 2007-nov-30  gayno    - improved method for thinning gfs grids
> +!!                         added nam b-grids
> +!! 2014-sep-29  gayno    - add option to read lat,lon and mask
> +!!                         data in grib2.
> +!!
> +!! usage: call read_mdl_grid_info
> +!!
> +!!   input argument list:  n/a
>
> Remove any documentation element that says n/a or none. It is not
> conveying information.
> ------------------------------
>
> In sorc/emcsfc_snow2mdl.fd/model_grid.f
> <https://github.com/NOAA-EMC/UFS_UTILS/pull/348#discussion_r581401056>:
>
> > +!! condition codes: all fatal
> +!!   76 - bad open/read gfs "lonsperlat" file
> +!!   79 - unrecognized model grid
> +!!   80 - bad open model latitude file
> +!!   81 - bad read of model latitude grib 1 header
> +!!   82 - bad read of model latitude data
> +!!   83 - bad open model longitude file
> +!!   82 - bad read of model longitude data
> +!!   85 - bad open model landmask file
> +!!   86 - bad read of model landmask data
> +!!   90 - model latitude file not grib 1 or grib 2
> +!!   91 - model longitude file not grib 1 or grib 2
> +!!   92 - model landmask file not grib 1 or grib 2
> +!!
> +!! remarks: none.
> +!!
>
>
> There should not be a blank like, and the documentation block should be
> after "contains".
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/UFS_UTILS/pull/348#pullrequestreview-596809769>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ANDS4FQROFHILOWH2KLZPB3TAQMU3ANCNFSM4YC6U2SA>
> .
>


-- 

George W Vandenberghe

*IMSG* at NOAA/NWS/NCEP/EMC

5830 University Research Ct., Rm. 2141

College Park, MD 20740

George.Vandenberghe@noaa.gov

301-683-3769(work)  3017751547(cell)

GeorgeVandenberghe-NOAA commented:
I will fix Gwv.   I thought I had overwritten it with a description.  Can
draft PR be inspected?

On Tue, Feb 23, 2021 at 4:44 PM Edward Hartnett <notifications@github.com>
wrote:

> Please also correct the name of this PR and its description. In future, do
> not introduce a PR with a name "Gwv" always use something descriptive to
> the rest of the team. Also use draft pull request for PRs you are working
> on, which are not ready for review.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/UFS_UTILS/pull/348#issuecomment-784533506>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ANDS4FWNYJPV73VHHIIVR5DTAQOSVANCNFSM4YC6U2SA>
> .
>


-- 

George W Vandenberghe

*IMSG* at NOAA/NWS/NCEP/EMC

5830 University Research Ct., Rm. 2141

College Park, MD 20740

George.Vandenberghe@noaa.gov

301-683-3769(work)  3017751547(cell)

edwardhartnett commented:
A draft PR can be inspected, but it communicates that this is a work in progress. Some might want to inspect as you work, others might want to wait until it's complete.

It can't be merged as a draft.

GeorgeVandenberghe-NOAA commented:
Next one will be a draft.

On Tue, Feb 23, 2021 at 4:50 PM Edward Hartnett <notifications@github.com>
wrote:

> A draft PR can be inspected, but it communicates that this is a work in
> progress. Some might want to inspect as you work, others might want to wait
> until it's complete.
>
> It can't be merged as a draft.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/UFS_UTILS/pull/348#issuecomment-784536826>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ANDS4FTUD5MPGZLW6UBC4BDTAQPJTANCNFSM4YC6U2SA>
> .
>


-- 

George W Vandenberghe

*IMSG* at NOAA/NWS/NCEP/EMC

5830 University Research Ct., Rm. 2141

College Park, MD 20740

George.Vandenberghe@noaa.gov

301-683-3769(work)  3017751547(cell)

edwardhartnett commented on sorc/emcsfc_snow2mdl.fd/model_grid.f:
```diff
@@ -59,56 +59,41 @@ module model_grid
+!!   91 - model longitude file not grib 1 or grib 2
+!!   92 - model landmask file not grib 1 or grib 2
+!!
+!! remarks: none.
```
Remove all documentation sections which are just "none" or "N/A".

edwardhartnett commented on sorc/emcsfc_snow2mdl.fd/model_grid.f:
```diff
@@ -59,56 +59,41 @@ module model_grid
+!!                         added nam b-grids
+!! 2014-sep-29  gayno    - add option to read lat,lon and mask
+!!                         data in grib2.
+!! usage: call read_mdl_grid_info
```
Remove the entire usage line. Doxygen does this automatically.

edwardhartnett commented on sorc/emcsfc_snow2mdl.fd/model_grid.f:
```diff
@@ -59,56 +59,41 @@ module model_grid
+!! usage: call read_mdl_grid_info
+!! files:
+!!   inputs:
+!!     - model latitudes (grib 1 or grib 2)
```
Use param tags to document parameters.

edwardhartnett commented on sorc/emcsfc_snow2mdl.fd/model_grid.f:
```diff
@@ -59,56 +59,41 @@ module model_grid
+!!     - model longitudes (grib 1 or grib 2)
+!!     - model landmask (grib 1 or grib 2)
+!!     - number  pts per row, gfs grid (the "lonsperlat" file, ascii)
+!!  outputs: none
```
Remove lines which are just "none"

edwardhartnett commented on sorc/emcsfc_snow2mdl.fd/snowdat.f:
```diff
@@ -1,7 +1,7 @@
 !! @brief Read and qc afwa, nesdis/ims and autosnow snow data.
 !!
-!! @author gayno org: w/np2 @date 2005-dec-16
+!! @author gayno org: w/np2 @date 2005-dec-16 gwv
```
Remove your initials. Capitalize George's name.

edwardhartnett commented on sorc/emcsfc_snow2mdl.fd/snowdat.f:
```diff
@@ -96,41 +96,35 @@ module snowdat
-
  contains
-
+!>
```
Make this match the example.

!> Some one-sentence description.
!!
!! More sentences of desciption. 

Take out the name of the subprogram, the word "abstact:", captialize the sentences.

edwardhartnett commented on sorc/emcsfc_snow2mdl.fd/snowdat.f:
```diff
@@ -96,41 +96,35 @@ module snowdat
+!!   input:
+!!     - autosnow data, grib 2, unit=lugb
+!!
+!!   output: none
```
Remove this line.

edwardhartnett commented on sorc/emcsfc_snow2mdl.fd/snowdat.f:
```diff
@@ -96,41 +96,35 @@ module snowdat
+!! program history log:
+!! 2008-feb-04  gayno    - initial version
+!!
+!! usage: call readautosnow
```
Remove this line.

edwardhartnett commented on sorc/emcsfc_snow2mdl.fd/snowdat.f:
```diff
@@ -96,41 +96,35 @@ module snowdat
+!! remarks: autosnow data available only for southern hemis.
+!!          autosnow data is in grib 2.          
+!!
+!!
```
Remove one of these empty lines.
Remove "remark:" and replace it with @note . Capitalize sentences.

GeorgeVandenberghe-NOAA commented:
I thought I'd cleaned up some of these already. Others are misses.  Fixing.


On Wed, Feb 24, 2021 at 9:59 AM Edward Hartnett <notifications@github.com>
wrote:

> *@edwardhartnett* requested changes on this pull request.
> ------------------------------
>
> In sorc/emcsfc_snow2mdl.fd/model_grid.f
> <https://github.com/NOAA-EMC/UFS_UTILS/pull/348#discussion_r582026002>:
>
> > +!!  outputs: none
> +!!    condition codes: all fatal
> +!!   76 - bad open/read gfs "lonsperlat" file
> +!!   79 - unrecognized model grid
> +!!   80 - bad open model latitude file
> +!!   81 - bad read of model latitude grib 1 header
> +!!   82 - bad read of model latitude data
> +!!   83 - bad open model longitude file
> +!!   82 - bad read of model longitude data
> +!!   85 - bad open model landmask file
> +!!   86 - bad read of model landmask data
> +!!   90 - model latitude file not grib 1 or grib 2
> +!!   91 - model longitude file not grib 1 or grib 2
> +!!   92 - model landmask file not grib 1 or grib 2
> +!!
> +!! remarks: none.
>
> Remove all documentation sections which are just "none" or "N/A".
> ------------------------------
>
> In sorc/emcsfc_snow2mdl.fd/model_grid.f
> <https://github.com/NOAA-EMC/UFS_UTILS/pull/348#discussion_r582026346>:
>
> > @@ -59,56 +59,41 @@ module model_grid
>   real                           :: resol_mdl  ! in km
>
>   contains
> -
> +!>
> +!!
> +!! @author George Gayno org: w/np2 @date 2005-dec-16
> +!!
> +!! program history log:
> +!! 2005-dec-16  gayno    - initial version
> +!! 2007-nov-30  gayno    - improved method for thinning gfs grids
> +!!                         added nam b-grids
> +!! 2014-sep-29  gayno    - add option to read lat,lon and mask
> +!!                         data in grib2.
> +!! usage: call read_mdl_grid_info
>
> Remove the entire usage line. Doxygen does this automatically.
> ------------------------------
>
> In sorc/emcsfc_snow2mdl.fd/model_grid.f
> <https://github.com/NOAA-EMC/UFS_UTILS/pull/348#discussion_r582027088>:
>
> > @@ -59,56 +59,41 @@ module model_grid
>   real                           :: resol_mdl  ! in km
>
>   contains
> -
> +!>
> +!!
> +!! @author George Gayno org: w/np2 @date 2005-dec-16
> +!!
> +!! program history log:
> +!! 2005-dec-16  gayno    - initial version
> +!! 2007-nov-30  gayno    - improved method for thinning gfs grids
> +!!                         added nam b-grids
> +!! 2014-sep-29  gayno    - add option to read lat,lon and mask
> +!!                         data in grib2.
> +!! usage: call read_mdl_grid_info
> +!! files:
> +!!   inputs:
> +!!     - model latitudes (grib 1 or grib 2)
>
> Use param tags to document parameters.
> ------------------------------
>
> In sorc/emcsfc_snow2mdl.fd/model_grid.f
> <https://github.com/NOAA-EMC/UFS_UTILS/pull/348#discussion_r582027325>:
>
> > +!! @author George Gayno org: w/np2 @date 2005-dec-16
> +!!
> +!! program history log:
> +!! 2005-dec-16  gayno    - initial version
> +!! 2007-nov-30  gayno    - improved method for thinning gfs grids
> +!!                         added nam b-grids
> +!! 2014-sep-29  gayno    - add option to read lat,lon and mask
> +!!                         data in grib2.
> +!! usage: call read_mdl_grid_info
> +!! files:
> +!!   inputs:
> +!!     - model latitudes (grib 1 or grib 2)
> +!!     - model longitudes (grib 1 or grib 2)
> +!!     - model landmask (grib 1 or grib 2)
> +!!     - number  pts per row, gfs grid (the "lonsperlat" file, ascii)
> +!!  outputs: none
>
> Remove lines which are just "none"
> ------------------------------
>
> In sorc/emcsfc_snow2mdl.fd/snowdat.f
> <https://github.com/NOAA-EMC/UFS_UTILS/pull/348#discussion_r582027664>:
>
> > @@ -1,7 +1,7 @@
>  !> @file
>  !! @brief Read and qc afwa, nesdis/ims and autosnow snow data.
>  !!
> -!! @author gayno org: w/np2 @date 2005-dec-16
> +!! @author gayno org: w/np2 @date 2005-dec-16 gwv
>
> Remove your initials. Capitalize George's name.
> ------------------------------
>
> In sorc/emcsfc_snow2mdl.fd/snowdat.f
> <https://github.com/NOAA-EMC/UFS_UTILS/pull/348#discussion_r582028940>:
>
> >   contains
> -
> +!>
>
> Make this match the example.
>
> !> Some one-sentence description.
> !!
> !! More sentences of desciption.
>
> Take out the name of the subprogram, the word "abstact:", captialize the
> sentences.
> ------------------------------
>
> In sorc/emcsfc_snow2mdl.fd/snowdat.f
> <https://github.com/NOAA-EMC/UFS_UTILS/pull/348#discussion_r582029142>:
>
> > +!! subprogram:    readautosnow
> +!! @author  George Gayno          org: w/np2     date: 2008-Feb-04
> +!!
> +!! abstract:  read autosnow snow cover.
> +!!
> +!! program history log:
> +!! 2008-feb-04  gayno    - initial version
> +!!
> +!! usage: call readautosnow
> +!!
> +!!
> +!! files:
> +!!   input:
> +!!     - autosnow data, grib 2, unit=lugb
> +!!
> +!!   output: none
>
> Remove this line.
> ------------------------------
>
> In sorc/emcsfc_snow2mdl.fd/snowdat.f
> <https://github.com/NOAA-EMC/UFS_UTILS/pull/348#discussion_r582029235>:
>
> >   contains
> -
> +!>
> +!!
> +!! subprogram:    readautosnow
> +!! @author  George Gayno          org: w/np2     date: 2008-Feb-04
> +!!
> +!! abstract:  read autosnow snow cover.
> +!!
> +!! program history log:
> +!! 2008-feb-04  gayno    - initial version
> +!!
> +!! usage: call readautosnow
>
> Remove this line.
> ------------------------------
>
> In sorc/emcsfc_snow2mdl.fd/snowdat.f
> <https://github.com/NOAA-EMC/UFS_UTILS/pull/348#discussion_r582029857>:
>
> > +!!
> +!!
> +!! files:
> +!!   input:
> +!!     - autosnow data, grib 2, unit=lugb
> +!!
> +!!   output: none
> +!!
> +!! condition codes:  all fatal
> +!!   74 - bad open of autosnow file
> +!!   75 - bad read of autosnow file
> +!!
> +!! remarks: autosnow data available only for southern hemis.
> +!!          autosnow data is in grib 2.
> +!!
> +!!
>
> Remove one of these empty lines.
> Remove "remark:" and replace it with @note <https://github.com/note> .
> Capitalize sentences.
> ------------------------------
>
> In sorc/emcsfc_snow2mdl.fd/model_grid.f
> <https://github.com/NOAA-EMC/UFS_UTILS/pull/348#discussion_r582024471>:
>
> > @@ -1,7 +1,7 @@
>  !> @file
>  !! @brief Read in data defining the model grid.
>  !!
> -!! @author gayno org: w/np2 @date 2005-dec-16
> +!!@author gayno org: w/np2 @date 2005-dec-16
>
> Blank space before author tag.
> ------------------------------
>
> In sorc/emcsfc_snow2mdl.fd/model_grid.f
> <https://github.com/NOAA-EMC/UFS_UTILS/pull/348#discussion_r582025259>:
>
> > @@ -58,57 +58,53 @@ module model_grid
>   real, allocatable              :: lsmask_mdl_sav (:,:)
>   real                           :: resol_mdl  ! in km
>
> - contains
> +!>
>
> Remove the brief tag (the "@brief <https://github.com/brief>") but leave
> the sentence which documents the function/subroutine.
>
> See my example above.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/UFS_UTILS/pull/348#pullrequestreview-597565263>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ANDS4FTRK4KXW2YZXMSHCMTTAUH33ANCNFSM4YC6U2SA>
> .
>


-- 

George W Vandenberghe

*IMSG* at NOAA/NWS/NCEP/EMC

5830 University Research Ct., Rm. 2141

College Park, MD 20740

George.Vandenberghe@noaa.gov

301-683-3769(work)  3017751547(cell)

edwardhartnett commented on sorc/emcsfc_snow2mdl.fd/model_grid.f:
```diff
@@ -575,33 +572,23 @@ subroutine read_mdl_grid_info
  return
+ end
 
- end subroutine read_mdl_grid_info
```
Did you remove this line on purpose?

GeorgeVandenberghe-NOAA replied:
No.   Accident.   Compile failure identified the error but I thought I fixed it before committing and pushing.

added @note in place of remarks, and committed

edwardhartnett commented on sorc/emcsfc_snow2mdl.fd/model_grid.f:
```diff
@@ -575,33 +572,23 @@ subroutine read_mdl_grid_info
+!!
+!!
+!!
+!!
```
One empty line at most please.

GeorgeVandenberghe-NOAA replied:
cleaning this up

edwardhartnett commented on sorc/emcsfc_snow2mdl.fd/model_grid.f:
```diff
@@ -575,33 +572,23 @@ subroutine read_mdl_grid_info
 
+!>   clean up allocatable arrays  
+!!
+!!@ author   George Gayno          org: w/np2     date: 2005-Dec-16
```
Messed up author tag.

!! @author George Gayno @date 2005-Dec-16

No extra spaces, no colon after date.

GeorgeVandenberghe-NOAA replied:
Corrected author tag, spacing, and (in a previous push) the subroutine end statement

edwardhartnett reviewed (changes requested):
Getting closer...

edwardhartnett reviewed (changes requested):
Stop adding new files to this PR and concentrate on getting the ones you have correct.

I suspect you are not really reading my review comments, because I keep having to write the same things over and over.

Is there a reason for that? 

Do you feel this work is not worthy of your time and attention?

edwardhartnett commented:
@GeorgeVandenberghe-NOAA you need to focus on this task and pay attention to my review comments. We've gotten to the point where every comment I type, I remember already typing it 2, 3, 4, or 5 times before.

GeorgeVandenberghe-NOAA commented:
I am trying to proceed with others while waiting for reviews.

And no, this is my most important project esp  since it is revealing gross
deficiencies in my skills .

Typing from phone since NCO updated my machine and its taking awhile.







On Thursday, February 25, 2021, Edward Hartnett <notifications@github.com>
wrote:

> @GeorgeVandenberghe-NOAA <https://github.com/GeorgeVandenberghe-NOAA> you
> need to focus on this task and pay attention to my review comments. We've
> gotten to the point where every comment I type, I remember already typing
> it 2, 3, 4, or 5 times before.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/UFS_UTILS/pull/348#issuecomment-785846273>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ANDS4FVIWFP2RBVXIUJGNCDTAY4BRANCNFSM4YC6U2SA>
> .
>


-- 

George W Vandenberghe

*IMSG* at NOAA/NWS/NCEP/EMC

5830 University Research Ct., Rm. 2141

College Park, MD 20740

George.Vandenberghe@noaa.gov

301-683-3769(work)  3017751547(cell)

GeorgeVandenberghe-NOAA commented on sorc/emcsfc_snow2mdl.fd/snow2mdl.f:
```diff
@@ -811,37 +800,27 @@ subroutine interp
 
+!>  subprogram documentation block
+!!
+!! subprogram:   write_grib2
```
My own laptop is back so I can read these comments and address.

I have told various people asking for my help that this is more important and it will NEVER EVER get done if it is not addressed and it is therefore my most important project and we're in this mess because of previous shortcuts and postponements for "urgencies".   I think we're on the same page there.  I am also disappointed in the slow progress but
with improvements in my comprehension of the process I should progress much more rapidly.

GeorgeVandenberghe-NOAA replied:
!!subprogram write_grib2 doc removed.  !!Subprogram documentation block removed

GeorgeVandenberghe-NOAA commented on sorc/emcsfc_snow2mdl.fd/snow2mdl.f:
```diff
@@ -86,94 +86,83 @@ module snow2mdl
+!!   language: fortran 90
+!!   machine:  IBM WCOSS
+!!
+!!
```
Okay, will go through everything for multiple empty lines.  It probably happened when I took out NA and none lines and left !!

This is the first time you've mentioned author at the bottom of the docblock so I'll go through the routines I've done and fix  to conform

GeorgeVandenberghe-NOAA replied:
going through all code for multiple !!

GeorgeVandenberghe-NOAA commented on sorc/emcsfc_snow2mdl.fd/snow2mdl.f:
```diff
@@ -811,37 +800,27 @@ subroutine interp
 
  end subroutine interp
 
+!>  subprogram documentation block
```
Nope.  Missed that.

GeorgeVandenberghe-NOAA replied:
It had more but it was all NA and none and other excess information to be removed

GeorgeVandenberghe-NOAA replied:
Actually going back I found a nonempty docblock for write_grib2.   Repaired as described in other sections

edwardhartnett commented:
@GeorgeVandenberghe-NOAA something I always do when submitting a PR is immediately review my own code. That is, go to the "Files Changed 3" tab above, and review the changes to the code that you are submitting.

In this way I catch many of my own mistakes before anyone ever sees them, and, after all, this is every programmer's fondest wish.

GeorgeVandenberghe-NOAA commented on sorc/emcsfc_snow2mdl.fd/snow2mdl.f:
```diff
@@ -86,94 +86,83 @@ module snow2mdl
+!!      defined threshold, the depth is set to 0,
+!!      regardless of the afwa depth value.
+!!      
+!! attributes:
```
All attributes should now be gone.   No PR yet.. addressing other issues in previous attempts

GeorgeVandenberghe-NOAA commented on sorc/emcsfc_snow2mdl.fd/snow2mdl.f:
```diff
@@ -86,94 +86,83 @@ module snow2mdl
-
+!>   subprogram documentation block
+!!
+!!  @author George Gayno org: w/np2  @date 2005-Dec-16
```
Moved all author tags to bottom of docblocks and changed date to @date as needed in author tags

GeorgeVandenberghe-NOAA commented:
There remains an issue in snowdat.f and model_grid.f.  Doxygen docblocks above the CONTAINED subroutines are not getting included in the html output by doxygen.  Proceeding with other requested documentation while waiting for an answer to this.

edwardhartnett commented:
George, ignore the issue of the documentation blocks not being included in the outcome for this PR. Fix all remaining formatting issues, and we will circle back around to the missing documentation problem after the format changes have been merged.

edwardhartnett commented:
Your branch is out of date with the base branch, and that is typical, because the base branch changes while you are working on your branch, since others are also working on the code.

So make sure you update your fork with a fetch, and then update your branch by merging the latest develop branch to it.

GeorgeVandenberghe-NOAA commented:
Remaining repairs and actions taken

-------------------------------------------
You still have not addressed many of the comments I made in your open PR.

When I look at the files in your PR I see obvious doxygen mistakes:

Missing description of function:

**** Found this.. fixed

!>
!!
!! program history log:
!! 2005-dec-16  gayno    - initial version
!! 2007-nov-30  gayno    - improved method for thinning gfs grids
!!                         added nam b-grids
!! 2014-sep-29  gayno    - add option to read lat,lon and mask
!!                         data in grib2.

Remove documentation sections with "none" and the "attributes" section:

********* There were no attributes sections.  I found one remaining "none" in the source

!>    grib 1 writer                        
￼!!
￼!! abstract:  output snow cover and depth on the model grid
￼!!            in grib1 format.
￼!!
￼!! program history log:
￼!! 2005-dec-16  gayno    - initial version
￼!! 2014-sep-26  gayno    - rename as write_grib1 (was gribit)
￼!!
￼!!  files!
￼!!   output:
￼!!     - snow on model grid, grib 1, unit=lugb
￼!!
￼!! condition codes:
￼!!    57 - error writing model snow depth record
￼!!    58 - error writing model snow cover record
￼!!    59 - error opening model snow file
￼!!
￼!! remarks: none.
￼!!      
￼!!   @author George Gayno org: w/np2     @date  2005-Dec-16

Extra unnecessary spacing and hyphens around param tags, some param lines with 2 params instead of 1:

!! @param[in]  - kgds_data  - grib 1 grid description sect of data to be qcd
￼!! @param[in]  - snow_data  - snow cover to be qcd
￼!! @param[in]  - bitmap_data - bitmap of data to be qcd
￼!! @param[in]  - i/jdata     - i/j dimension of data to be qcd
￼!! @param[in]  - isrc        - flag indicating data source; 1-afwa depth, 2-ims cover

********************will proofread params for errors.  In progress

Uncapitalized and unpunctuated sentences everywhere.

_*************************Will proofread docblocks for these***

Can you please fix?

************The issue remains I can't get docblocks of CONTAINED subroutines to appear

GeorgeVandenberghe-NOAA commented:
for docblocks containing lines like

! condition codes:
!!    57 - error writing model snow depth record
!!    58 - error writing model snow cover record
!!    59 - error opening model snow file
!!

Do you want them turned into sentences  (capitalize first letter and period last space)?

edwardhartnett commented:
No, those are fine. But there's no reason for all the sentences to start with lower-case letters, or to lack punctuation.

GeorgeVandenberghe-NOAA commented:
I have been doing the following before commits and pushes

rm -rf us
git clone git@github.com:GeorgeVandenberghe-NOAA/UFS_UTILS.git us
  cd us
  git remote add upstream git@github.com:NOAA-EMC/UFS_UTILS.git
  git fetch upstream
  git checkout gwv

This should put my local branch gwv on mars in sync with the the NOAA-EMC
repository and the commit/push

^^ reapply updates

git commit -a
git push -u origin gwv

gets the GeorgeVandenberghe-NOAA fork sync.ed

So they shouldn't be out of sync.



On Fri, Feb 26, 2021 at 12:17 PM Edward Hartnett <notifications@github.com>
wrote:

> Your branch is out of date with the base branch, and that is typical,
> because the base branch changes while you are working on your branch, since
> others are also working on the code.
>
> So make sure you update your fork with a fetch, and then update your
> branch by merging the latest develop branch to it.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/UFS_UTILS/pull/348#issuecomment-786778957>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ANDS4FXG5LYDLSN6SCQXHOTTA7JRBANCNFSM4YC6U2SA>
> .
>


-- 

George W Vandenberghe

*IMSG* at NOAA/NWS/NCEP/EMC

5830 University Research Ct., Rm. 2141

College Park, MD 20740

George.Vandenberghe@noaa.gov

301-683-3769(work)  3017751547(cell)

GeorgeVandenberghe-NOAA commented:
the matching of module docblock attributes to variable definitions with !< doxygen comments in snowdat.f  is proving to be extremely slow and error prone. I've only gotten six.. keep corrupting the original source and redoing.  Is there an editing trick? I am using vi

At this rate I don't think I'll be done this afternoon.

GeorgeVandenberghe-NOAA commented:
Also encountering these

logical  bad_afwa_sh, bad_nesdis, bad_afwa_global

do I split into 
logical : $name
for each one?

edwardhartnett commented:
Yes, when multiple variables are defined on a line, I break them into their own line:

`integer :: x, y
`
becomes

```
integer :: x !< Documentation of X
integer :: y !< Documentation of Y
```

edwardhartnett reviewed (approved):
These look good. We will circle back around and fill in the missing docs...
